ee/api/test/test_team_memberships.py

```diff
@@ -0,0 +1,380 @@
+from rest_framework import status
+
+from ee.api.test.base import APILicensedTest
+from ee.models.explicit_team_membership import ExplicitTeamMembership
+from posthog.models import OrganizationMembership, Team, User
+
+
+class TestTeamMembershipsAPI(APILicensedTest):
+    def setUp(self):
+        super().setUp()
+        self.organization.per_project_access = True
+        self.organization.save()
+
+    def test_add_member_as_org_owner_allowed(self):
+        self.organization_membership.level = OrganizationMembership.Level.OWNER
+        self.organization_membership.save()
+
+        new_user: User = User.objects.create_and_join(self.organization, "rookie@posthog.com", None)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 0)
+
+        response = self.client.post("/api/projects/@current/explicit_members/", {"user_uuid": new_user.uuid})
+        response_data = response.json()
+
+        self.assertDictContainsSubset(
+            {"effective_level": ExplicitTeamMembership.Level.MEMBER, "level": ExplicitTeamMembership.Level.MEMBER,},
+            response_data,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 1)
+
+    def test_add_member_as_org_admin_allowed(self):
+        self.organization_membership.level = OrganizationMembership.Level.ADMIN
+        self.organization_membership.save()
+
+        new_user: User = User.objects.create_and_join(self.organization, "rookie@posthog.com", None)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 0)
+
+        response = self.client.post("/api/projects/@current/explicit_members/", {"user_uuid": new_user.uuid})
+        response_data = response.json()
+
+        self.assertDictContainsSubset(
+            {"effective_level": ExplicitTeamMembership.Level.MEMBER, "level": ExplicitTeamMembership.Level.MEMBER,},
+            response_data,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 1)
+
+    def test_add_member_as_org_member_forbidden(self):
+        self.organization_membership.level = OrganizationMembership.Level.MEMBER
+        self.organization_membership.save()
+
+        new_user: User = User.objects.create_and_join(self.organization, "rookie@posthog.com", None)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 0)
+
+        response = self.client.post("/api/projects/@current/explicit_members/", {"user_uuid": new_user.uuid})
+        response_data = response.json()
+
+        self.assertDictEqual(
+            self.permission_denied_response("You don't have sufficient permissions in this project."), response_data
```

Comment:
"[...] for this project or organization."

Reply:
Current wording seems the clearest to me, in that elevated organization permissions aren't actually needed – having elevated project-specific perms is enough.